ssl/record/rec_layer_s3.c

```diff
@@ -268,11 +268,15 @@ int ssl3_read_n(SSL *s, size_t n, size_t max, int extend, int clearold,
         return -1;
     }
 
-    /* We always act like read_ahead is set for DTLS */
-    if (!s->rlayer.read_ahead && !SSL_IS_DTLS(s))
+    /*
+     * Ktls always reads full records.
+     * Also, we always act like read_ahead is set for DTLS.
+     */
+    if (!BIO_get_ktls_recv(s->rbio) && !s->rlayer.read_ahead
```

Comment:
This is used to avoid splitting the read between header and data in `ssl3_get_record`. Instead, using this, we read the maximum size on the call to read the header.

For ktls reading the header alone doesn't make any sense...